asyncpg/types.py

```diff
@@ -228,6 +228,28 @@ def as_string(self):
 
         return s.strip()
 
+    def to_int(self, bitorder):
+        x = int.from_bytes(self._bytes, byteorder='big')
+        x >>= -self._bitlength % 8
+        if bitorder == 'little':
+            return int(bin(x)[:1:-1].ljust(self._bitlength, '0'), 2)
+        elif bitorder == 'big':
+            return x
+        else:
+            raise ValueError("bitorder must be either 'little' or 'big'")
+
+    @classmethod
+    def from_int(cls, x, length, bitorder):
```

Comment:
`length` should be an upper limit (like in `int.to_bytes`), not an actual bitstring length specification. The actual length can be unambiguously determined from the passed value.  I would also make it optional.

Reply:
I'm confused by this. The `length` argument of `int.to_bytes()` is not an upper limit, it's an actual byte length. And, as I understand it, it's not possible to unambiguously determine a bit length from a python integer, because there may be leading zeros (that aren't reflected in anything like `int.bit_length()`).

Reply:
Ah, okay. I didn't guess that Postgres would automatically convert shorter bits to longer bits. Will do.

Reply:
Okay, turns out length is necessary; asyncpg throws a StringDataLengthMismatchError instead of automatically padding with zeros. Which is just as well, cuz big-endian/little-endian/left-pad/right-pad is kinda confusing here :P